File: app/src/main/java/tw/y_studio/ptt/UI/ClickFix.java
package tw.y_studio.ptt.UI;

import java.util.Date;

public class ClickFix {

    public ClickFix(long defaultTime) {
        this.defaultTime = defaultTime;
    }

    public ClickFix() {}

    private long lastClickTime = 0;
    private long defaultTime = 500L;

    public boolean isFastDoubleClick() {
        return isFastDoubleClick(defaultTime);
    }

    public boolean isFastDoubleClick(long time2) {
        long time = new Date().getTime();
        // long timeD = time - lastClickTime;
        if (Math.abs(time - lastClickTime) < time2) {
            // lastClickTime = time;
            return true;
        }
        lastClickTime = time;
        return false;
    }
}

File: app/src/main/java/tw/y_studio/ptt/ui/ClickFix.kt
package tw.y_studio.ptt.ui

import java.util.*
import kotlin.math.abs

class ClickFix(private var defaultTime: Long = 500L) {

    private var lastClickTime: Long = 0
    val isFastDoubleClick: Boolean
        get() = isFastDoubleClick(defaultTime)

    fun isFastDoubleClick(time2: Long): Boolean {
        val time = Date().time
        if (abs(time - lastClickTime) < time2) {
            return true
        }
        lastClickTime = time
        return false
    }
}
